lib/rust/ensogl/component/grid-view/src/visible_area.rs

```diff
@@ -38,18 +38,39 @@ pub fn visible_columns(
     col_count: usize,
     column_widths: &ColumnWidths,
 ) -> Range<Col> {
-    let column_left = |idx: usize| idx as f32 * entry_size.x + column_widths.pos_offset(idx);
+    let entry_width = entry_size.x;
+    let column_left = |idx: usize| idx as f32 * entry_width + column_widths.pos_offset(idx);
     let column_right = |idx: usize| column_left(idx + 1);
     let index = |(idx, _): (usize, _)| idx;
-
-    let visible = |(_, x): &(_, f32)| v.left < *x;
-    let mut right_borders = (0..col_count).map(|idx| (idx, column_right(idx)));
-    let first_visible = right_borders.find(visible).map(index).unwrap_or(col_count);
-
+    let clamp = |idx: isize| idx.clamp(0, col_count as isize) as Col;
+
+    let left_guess = clamp((v.left / entry_width).floor() as isize);
+    let first_visible = {
```

Comment:
Add an explanation why we do such guesses.

Reply:
Added